src/api/baseapi.cpp

```diff
@@ -925,6 +925,25 @@ Pix *TessBaseAPI::GetInputImage() {
   return tesseract_->pix_original();
 }
 
+// Grayscale normalization (preprocessing)
+bool TessBaseAPI::NormalizeImage(int mode){
+  if (!GetInputImage()){
+    tprintf("Please use SetImage before applying the image pre-processing steps.");
+    return false;
+  }
+  if (mode == 1) {
+    SetInputImage(thresholder_->GetPixNormRectGrey());
+    thresholder_->SetImage(GetInputImage());
+  } else if (mode == 2) {
+    thresholder_->SetImage(thresholder_->GetPixNormRectGrey());
+  } else if (mode == 3) {
+    SetInputImage(thresholder_->GetPixNormRectGrey());
+  } else {
+    return false;
+  }
```

Comment:
Some considerations regarding where this should be placed in the code base:
1. Using a separate entry point `NormalizeImage` called in `ProcessPages` instead of merely modifying the thresholder prevents applying this on any PSM other than full pages. And on the API, you would need to add `NormalizeImage` to the calling code instead of merely setting the configuration parameter.
2. Recognition (`SetupForRecognition` → `BestPix`) does not always use `pix_original_`: after `SetRectangle()`, it uses `pix_grey_` or even `pix_binary_`.
3. Layout analysis mostly uses `pix_binary_`, but `LineFinder` also tries to use `pix_grey_` and `pix_thresholds_`.
4. DPI information (which influences LA in various ways) is taken from `pix_` (i.e. the thresholder's `SetImage`), and that might not work on the output of Leptonica because the metadata might be lost. We still have fallback DPI estimation (which is based on the CC statistics from `pix_binary_`), but that might not be as accurate.